### DISCUSSION TITLE
Conditionnal runtime export in app directory ?

### DISCUSSION THREAD

**Discussion by Julien-Goletto** [vercel/next.js#49195] [Help]
### Summary

While recreating on my own [Guillermo Rauch's idea](https://how-is-this-not-illegal-sooty.vercel.app/) I've been stuck on a silly problem.

```tsx
export const runtime = 'edge';
```

Prevents my code to be functionnal in local development mode, yet it is perfectly running in production on Vercel.
I tried a workaround with a conditionnal export, which is not working (configuration declarations have to be statically analyzable) :
```tsx
export const runtime = (process.env.NODE_ENV === 'development')? 'nodejs' : 'edge';
```

Does anyone has an idea on how could it work properly in both dev and prod modes ?
I'm kinda new on these new features, I may be not quite understand some of their details.

[My deployed project working on Vercel](https://how-is-this-not-illegal-sooty.vercel.app/)

### Additional information

**Project architecture :**
```
--app
    |
    |-- layout.tsx
    |-- loading.tsx
    |-- page.tsx
--components
    |
    |-- PokemonGrid.tsx
    |-- PokemonCard.tsx
```

**page.tsx**
```tsx
import { PokemonGrid } from '~/components/PokemonGrid';
import { dbQuery } from '~/utils/dbQuery';
import type { Pokemon } from '~/types/Pokemon';

export default async function Home() {
  const pokemons: Pokemon[] = await dbQuery(`SELECT * FROM pokemon_with_types ORDER BY RANDOM() LIMIT 12;`);
  return (
    <main className="container grow mx-auto pb-6 flex flex-col items-center justify-between text-slate-700">
      <PokemonGrid pokemons={pokemons}/>
    </main>
  );
}

// export const runtime = (process.env.NODE_ENV === 'development')? 'nodejs' : 'edge';
export const runtime = 'edge';
export const dynamic = "force-dynamic";
```

**Comment by icyJoseph** [comment]
What error are you seeing? AFAIK you should be able to work with runtime edge locally. However, once we include the `"@vercel/postgres"`, perhaps some contract/expectation is broken? Also just starting to explore these new Vercel offerings, so I might be missing something.
